libmamba/src/specs/version_spec.cpp

```diff
@@ -264,12 +264,7 @@ fmt::formatter<mamba::specs::VersionPredicate>::format(
             }
             if constexpr (std::is_same_v<Op, VersionPredicate::compatible_with>)
             {
-                out = fmt::format_to(
-                    out,
-                    "{}{}",
-                    VersionSpec::compatible_str,
-                    pred.m_version.str(op.level)
```

Comment:
Note to reviewer: this line cause the bug as the internals rely on the valid assumption that:

```c++
std::string match_spec_as_str = "numpy~=1.26"; // for instance.
match_spec_as_str == MatchSpec::parse(match_spec_as_str).value().str()
```

which was not verified here.

I think we need to add tests cases (in another PR) to test that this is valid for many `MatchSpec`.